### PR TITLE
fixed get data from archiver

### PR DESCRIPTION
**Pull request by jjanice**
switched to get data from archiver in XfelDispTest2.py then fixed the function in arPullGlob (getValuesOverTimeRange) to use lcls-tools/data_analysis/archiver to get the data then load up the variables as the getValuesOverTimeDummy did.
Also changed makList to create the PV prefixes (ACCL:L0B:0110:, ...) from lcls_tools/devices/scLinac. Lastly changed the default date/time on the display to be yesterday and the day before so we don't have to type the year. In the process some of the colors got changed magically.

### CODE REVIEW

File: XfelDispTest2.py [hunk 1]
```diff
@@ -93,9 +93,9 @@ def setGOVal(self,ac, ac2):
         EndTimeForArchiver = val2.toPyDateTime().strftime('%m/%d/%Y %H:%M:%S')
         #print(StTimeForArchiver)
         if EndTimeForArchiver > StTimeForArchiver:   # Checks that the End time is after the Start time
-            #arPullGlob.getValuesOverTimeRange(StTimeForArchiver, EndTimeForArchiver)
+            arPullGlob.getValuesOverTimeRange(StTimeForArchiver, EndTimeForArchiver)
             #  Use Dummy value generator for now
-            arPullGlob.getValuesOverTimeDummy(StTimeForArchiver, EndTimeForArchiver)
```
Comment (lisazacarias) [inline XfelDispTest2.py:98]: Could you put these in a conditional? Like, have a global bool DEBUG and have something like
`if DEBUG:
    arPullGlob.getValuesOverTimeDummy(StTimeForArchiver, EndTimeForArchiver)
else:
    arPullGlob.getValuesOverTimeRange(StTimeForArchiver, EndTimeForArchiver)
`

File: arPullGlob.py [hunk 2]
```diff
@@ -22,108 +24,71 @@
...
 
-    def getValuesOverTimeRange(self, strtTim, endTim, timeInterval=None):
+def getValuesOverTimeRange(strtTim, endTim, timeInterval=None):
  #       # type: (List[str], datetime, datetime, int) -> Dict[str, Dict[str, List[Union[datetime, str]]]]
-        global resultsFB, resultsIntlk, resultsRfRdy, startTime, endTime
-        pvList=[]
-        cavPvRfRdy={}
-        cavPvFB={}
-        cavPvIntlk={}
+    global resultsFB, resultsIntlk, resultsRfRdy, startTime, endTime
```
Comment (lisazacarias) [inline arPullGlob.py:43]: Just curious about the global tag here. I think I would personally prefer things like `RESULTS_FB = None` at the top of the file if they're being treated as globals (vs being defined here)

File: arPullGlob.py [hunk 3]
```diff
@@ -22,108 +24,71 @@
...
-                    cavPvRfRdy.append(str(pv)+codeFlt[2])
-            for pv in cavPv:
-                for i in range(3):
-                    pvList.append(str(pv) + codeFlt[i])
-            
-            url = self.url_formatter.format(SUFFIX=RANGE_RESULT_SUFFIX)
+    archiver=Archiver("lcls")
+
+    #global cavPvFB, cavPvIntlk, cavPvRfRdy
+    if (resultsFB == {}) or (strtTim != (time.mktime(datetime.datetime.strptime(strtTim, '%m/%d/%Y %H:%M:%S' ).timetuple()))) or (endTime != int(time.mktime(datetime.datetime.strptime(endTim, '%m/%d/%Y %H:%M:%S' ).timetuple()))):
```
Comment (lisazacarias) [inline arPullGlob.py:54]: This line is super long, it would help to break it up a bit for readability

File: arPullGlob.py [hunk 4]
```diff
@@ -22,108 +24,71 @@
...
-                           result = {"times": [], "values": []}
-                           for datum in element[u'data']:
-                               result["times"].append(datum[u'secs'])
-                               result["values"].append(datum[u'val'])
-    
-                           results[pv] = result["values"],result["times"]
-    
+        for pv in pvList:
+            print('PV: {}'.format(pv))
+            archiverData=archiver.getValuesOverTimeRange([pv],startTime,endTime)
```
Comment (lisazacarias) [inline arPullGlob.py:74]: I think you can probably pull this call out of the for loop as `getValuesOverTimeRange(pvList, startTime, endTime)` unless I'm misreading it

File: arPullGlob.py [hunk 5]
```diff
@@ -192,7 +159,9 @@ def cmFlts():
     stDHi=[]
 
     aaaa=list(resultsIntlk.keys())
```
Comment (lisazacarias) [inline arPullGlob.py:161]: a general TODO re: variable names hahahaha

File: arPullGlob.py [hunk 6]
```diff
@@ -270,9 +239,11 @@ def CmStats():
     stDHi=[]
     
     pvList = makList()
-    codeFlt = ["0:FB_SUM", "0:RFS:INTLK_FIRST", "0:RFREADYFORBEAM"]
+#for when sonya fixes it
+#    codeFlt = ["FB_SUM", "RFS:INTLK_FIRST", "RFREADYFORBEAM"]
+    codeFlt = ["PHAFB_SUM", "RFS:INTLK_FIRST", "RFREADYFORBEAM"]
     for pv in pvList:
-        CavFltDat= (resultsIntlk[(str(pv)+ codeFlt[1])]['times'])
+        CavFltDat= (resultsIntlk[(str(pv)+ codeFlt[1])]['times']) #was ['times'] not [1]
```
Comment (lisazacarias) [inline arPullGlob.py:246]: Comments above the line are preferable to comments after

File: arPullGlob.py [hunk 7]
```diff
@@ -369,7 +343,7 @@ def cavDatCnt(caVs, sT, fiN):
             bozo["CommFlt"] = clwn.count(8)
             bozo["SSAFlt"] = clwn.count(16)
             bozo["Quench"] = clwn.count(32)
-            bozo["Clips"] = len(resultsFB[pvFB]["values"])
+            bozo["Clips"] = len(resultsFB[pvFB]['values']) # was ["values"]
```
Comment (lisazacarias) [inline arPullGlob.py:346]: Oh, bozo... Eventually when we figure out what this is, we want to rename it hahaha

**Review by lisazacarias — APPROVED** [review]

File: XfelDispTest2.py (hunk 1 above)
Comment (jjanice) [reply XfelDispTest2.py:98]: agreed. Next edit.

File: arPullGlob.py (hunk 2 above)
Comment (jjanice) [reply arPullGlob.py:43]: Agreed. Not sure why they're globals, trying to suss that out now.

File: arPullGlob.py (hunk 3 above)
Comment (jjanice) [reply arPullGlob.py:54]: Agreed. I don't know what half of it means. He's checking the formatting of the time variables? There's gotta be a better way.

File: arPullGlob.py (hunk 4 above)
Comment (jjanice) [reply arPullGlob.py:74]: I'd like to run some speed tests. On matlab, surprisingly, calling get-archiver-data 3000 times is faster than calling it once with a list of 3000 pvs. I'd like to make sure the python code doesn't have the same "feature."

File: arPullGlob.py (hunk 5 above)
Comment (jjanice) [reply arPullGlob.py:161]: Indeed.  I just wanted to get it working.

File: arPullGlob.py (hunk 6 above)
Comment (jjanice) [reply arPullGlob.py:246]: left over from debugger stage. I can delete.

**Pull request merged**